From foo@baz Thu Sep 13 09:19:05 PDT 2012
Date: Thu, 13 Sep 2012 09:19:05 -0700
To: Greg KH <gregkh@linuxfoundation.org>
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Subject: USB: serial: move usb_serial_debug_data to use %*ph

Now that we have a printk modifier for data streams, use it instead of
rolling our own.

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

---
 include/linux/usb/serial.h |   12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

--- a/include/linux/usb/serial.h
+++ b/include/linux/usb/serial.h
@@ -360,15 +360,9 @@ static inline void usb_serial_debug_data
 					 const char *function, int size,
 					 const unsigned char *data)
 {
-	int i;
-
-	if (debug) {
-		dev_printk(KERN_DEBUG, dev, "%s - length = %d, data = ",
-			   function, size);
-		for (i = 0; i < size; ++i)
-			printk("%.2x ", data[i]);
-		printk("\n");
-	}
+	if (debug)
+		dev_printk(KERN_DEBUG, dev, "%s - length = %d, data = %*ph\n",
+			   function, size, size, data);
 }
 
 /* Use our own dbg macro */
